### PR TITLE
Improve deployment documentation

### PR DESCRIPTION
**Pull request by BenjaminPelletier**
This PR adds a considerable level of detail to our deployment documentation based on lessons learned from following the deployment procedure, including documenting variable values for example main.jsonnet & spec.json files in the deployment README.  It also exposes some configuration parameters (JWKS and public key files) in the example deployment configuration.

I welcome suggestions on how to satisfy the current `tk show --dangerous-allow-redirect build/deploy/examples/minimum` presubmit check.  It seems like minimum.jsonnet can't actually have a minimum set of values to generate a reasonable deployment configuration, so my baseline suggestion is that we should remove that presubmit check.

### CODE REVIEW

File: build/README.md [hunk 1]
```diff
@@ -14,188 +16,380 @@ To run a local DSS instance for testing, evaluation, or development, see [dev/st
...
+      - Under Frontend for Details, the Certificate column for HTTPS protocol
+        will have an icon next to it which will change to a green checkmark when
+        provisioning is complete.
+      - Click on the certificate link to see provisioning progress.
+      - If everything indicates OK and you still receive a cipher mismatch error
+        message when attempting to visit /healthy, wait an additional 5 minutes
+        before attempting to troubleshoot further.
+
+1.  If joining an existing cluster, share your CRDB node addresses with the
+    operator of the existing cluster.  They will add these node addresses to
```
Comment (BenjaminPelletier) [inline build/README.md:341]: @steeling or @charlie-pisuraj please verify this is what should happen -- basically, when a new DSS instance joins a DSS Region, the existing instances should be updated to point to the new instance's CRDB nodes, right?  Assuming so, is this the right way to do that?

Comment (steeling) [reply build/README.md:341]: they'll also need to update the certificates provided to their cockroachdb backends to contain both certs

File: build/README.md [hunk 2]
```diff
@@ -14,188 +16,380 @@ To run a local DSS instance for testing, evaluation, or development, see [dev/st
...
-1.  Use the `make-certs.py` script in this directory to create certificates for
+your chosen provider.
+
+If you can augment this documentation with specifics for another cloud provider,
+a PR to that effect would be greatly appreciated.
+
+1.  Create a new Kubernetes cluster. We recommend a new cluster for each DSS
+    instance.  A reasonable cluster name might be `dss-us-prod`,
+    `dss-ca-staging`, `dss-mx-integration`, etc.  The name of this cluster
+    will be referred to as CLUSTER_CONTEXT for the remainder of this document.
```
Comment (steeling) [inline build/README.md:109]: The cluster context is likely not the name of the cluster, but some longer key generated by kubectl

File: build/README.md [hunk 3]
```diff
@@ -14,188 +16,380 @@ To run a local DSS instance for testing, evaluation, or development, see [dev/st
...
-      DNS entries with a common hostname suffix, ie: `*.db.interuss.com`
-
-1.  Use the `make-certs.py` script in this directory to create certificates for
+your chosen provider.
+
+If you can augment this documentation with specifics for another cloud provider,
+a PR to that effect would be greatly appreciated.
+
+1.  Create a new Kubernetes cluster. We recommend a new cluster for each DSS
+    instance.  A reasonable cluster name might be `dss-us-prod`,
```
Comment (steeling) [inline build/README.md:107]: nit: why are we providing naming suggestions? If we are we might want to put more thought into the suggestions themselves. I would likely recommend including the availability zone in the name as well

File: build/README.md (hunk 1 above)
Comment (BenjaminPelletier) [reply build/README.md:341]: Yep, as described on line 196 above?

File: build/README.md (hunk 3 above)
Comment (BenjaminPelletier) [reply build/README.md:107]: To illustrate good practice and imply a set of rules a reasonable name should follow without actually listing out those rules -- we want someone to be able to follow these instructions with minimal interpretation or additional knowledge necessary and still end up with a good DSS instance.

Including zone sounds good; would you recommend `dss-us-us-east4-a-prod` or `dss-us-east4-a-prod` or `dss-us-prod-us-east-4a` or something else?

File: build/README.md (hunk 2 above)
Comment (BenjaminPelletier) [reply build/README.md]: Done; thanks.

File: build/README.md (hunk 1 above)
Comment (steeling) [reply build/README.md:343]: yup, missed that

File: build/README.md (hunk 3 above)
Comment (steeling) [reply build/README.md:107]: anything unique to the AZ sounds good to me. I'm partial to short names like `us-e4a`, resulting in `dss-prod-us-e4a`

Comment (BenjaminPelletier) [reply build/README.md]: Sounds good; done, thanks.

**Pull request merged**